R/diet.R

```diff
@@ -28,33 +28,47 @@
 #'
 #' @param params A \linkS4class{MizerParams} object
 #' @param n A matrix of species abundances (species x size).
+#' @param n_pp A vector of the resource abundance by size
 #' @param n_other A list of abundances for other dynamical components of the
-#'   ecosystem
 #' @param proportion If TRUE (default) the function returns the diet as a
 #'   proportion of the total consumption rate. If FALSE it returns the
 #'   consumption rate in grams per year.
 #'
 #' @return An array (predator species  x predator size x
-#'   (prey species + resources) )
+#'   (prey species + resource + other components) )
 #' @export
 #' @family summary functions
 #' @concept summary_function
-#' @seealso [plotDietMR()]
-
-
-
-getDietMR <- function (params, n = initialN(params),
-                       n_other = initialNOther(params), proportion = TRUE)
+#' @seealso [plotDiet()]
+#' @examples
+#' diet <- getDiet(NS_params)
+#' str(diet)
+getDiet <- function (params, n = initialN(params), n_pp = initialNResource(params),
+                     n_other = initialNOther(params), proportion = TRUE)
 {
     params <- validParams(params)
     species <- params@species_params$species
     no_sp <- length(species)
     no_w <- length(params@w)
     no_w_full <- length(params@w_full)
-    no_other <- dim(n_other$MR)[1]
-    other_names <- rownames(n_other$MR)
 
-    assert_that(identical(dim(n), c(no_sp, no_w)), dim(n_other$MR)[2] == no_w_full)
+    if(is.null(getComponent(params, "MR"))) # in this case there are no additional background so should have npp only
```

Comment:
I find this difficult to get my head around. Can we not simply always pass control to the mizer function when a mizerMR function is called with a params object that is missing multiple components, so that from then on the code only needs to deal with the case of multiple resources?